src/validation.h

```diff
@@ -524,6 +533,21 @@ class CChainState {
 
     bool LoadBlockIndex(const Consensus::Params& consensus_params, CBlockTreeDB& blocktree) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
+    /**
+     * Update the on-disk chain state.
+     * The caches and indexes are flushed depending on the mode we're called with
+     * if they're too large, if it's been a while since the last write,
+     * or always and in all cases if we're in prune mode and are deleting files.
+     *
+     * If FlushStateMode::NONE is used, then FlushStateToDisk(...) won't do anything
+     * besides checking if we need to prune.
+     */
+    bool FlushStateToDisk(
```

Comment:
In commit 9147f28ec0 refactoring: FlushStateToDisk -> CChainState


It seems weird to move a method that was `static` in validation.cpp to the header file as a public member function. At the very least it should be private. (However, I'd prefer if in the same commit the `FlushStateToDisk` and `PruneAndFlush` were moved to a public method in CChainState.)

Reply:
Agree with your general concern, but usage of `FlushStateToDisk` happens in places that won't work with `private` like `AcceptToMemoryPoolWithTime` and so if we made it private, we'd have to attach a bunch of `friend` statements which seems kind of gross.

Reply:
Hmm, makes sense